oidc-authenticator/src/main/java/org/xwiki/contrib/oidc/auth/internal/OIDCUserManager.java

```diff
@@ -304,10 +308,129 @@ private Principal updateUser(IDTokenClaimsSet idToken, UserInfo userInfo) throws
+    protected void addUserToXWikiGroup(String xwikiUserName, String groupName, XWikiContext context)
+    {
+        try {
+            BaseClass groupClass = context.getWiki().getGroupClass(context);
+
+            // Get document representing group
+            XWikiDocument groupDoc = context.getWiki().getDocument("XWiki." + groupName, context);
+
+            this.logger.info("Adding user [{}] to xwiki group [{}]. groupClass: [{}]; groupDoc: [{}]", xwikiUserName, groupName, groupClass, groupDoc);
+
+            synchronized (groupDoc) {
+                // Make extra sure the group cannot contain duplicate (even if this method is not supposed to be called
+                // in this case)
+                List<BaseObject> xobjects = groupDoc.getXObjects(groupClass.getDocumentReference());
+                if (xobjects != null) {
+                    for (BaseObject memberObj : xobjects) {
+                        if (memberObj != null) {
+                            String existingMember = memberObj.getStringValue(XWIKI_GROUP_MEMBERFIELD);
+                            if (existingMember != null && existingMember.equals(xwikiUserName)) {
+                                this.logger.warn("User [{}] already exist in group [{}]", xwikiUserName,
+                                        groupDoc.getDocumentReference());
+                                return;
+                            }
+                        }
+                    }
+                }
+
+                // Add a member object to document
+                BaseObject memberObj = groupDoc.newXObject(groupClass.getDocumentReference(), context);
+                Map<String, String> map = new HashMap<String, String>();
+                map.put(XWIKI_GROUP_MEMBERFIELD, xwikiUserName);
+                groupClass.fromMap(map, memberObj);
+
+                // If the document is new, set its content
+                if (groupDoc.isNew()) {
+                    groupDoc.setSyntax(Syntax.XWIKI_2_0);
+                    groupDoc.setContent("{{include reference='XWiki.XWikiGroupSheet' /}}");
+                }
+
+                // Save modifications
+                context.getWiki().saveDocument(groupDoc, context);
+            }
+
+            this.logger.info("Finished adding user [{}] to xwiki group [{}]", xwikiUserName, groupName);
+        } catch (Exception e) {
+            this.logger.error("Failed to add a user [{}] to a group [{}]", new Object[] {xwikiUserName, groupName, e});
+        }
+    }
+
+    /**
+     * Synchronize user XWiki membership with it's LDAP membership.
+     *
+     * @param xwikiUserName the name of the user.
+     * @param userDN the LDAP DN of the user.
+     * @param groupMappings the mapping between XWiki groups names and LDAP groups names.
+     * @param context the XWiki context.
+     * @throws XWikiException error when synchronizing user membership.
+     */
+    public void syncGroupsMembership(String xwikiUserName, List<String> groupMappings,
```

Comment:
It should be more explicit that this method is only working with groups coming from a XWiki based provider. Just `syncXWikiGroupsMembership` would be fine with me.